src/neo/Ledger/Blockchain.cs

```diff
@@ -289,26 +289,44 @@ private void OnFillMemoryPool(IEnumerable<Transaction> transactions)
             Sender.Tell(new FillCompleted());
         }
 
-        private RelayResultReason OnNewBlock(Block block)
+        private void OnInventory(IInventory inventory, bool relay = true)
+        {
+            RelayResult rr = new RelayResult
+            {
+                Inventory = inventory,
+                Result = inventory switch
+                {
+                    Block block => OnNewBlock(block),
+                    Transaction transaction => OnNewTransaction(transaction),
+                    ConsensusPayload payload => OnNewConsensus(payload),
+                    _ => VerifyResult.Unknown
+                }
+            };
+            if (relay && rr.Result == VerifyResult.Succeed)
+                system.LocalNode.Tell(new LocalNode.RelayDirectly { Inventory = inventory });
```

Comment:
Maybe we can reuse `rr` here, and avoid to create a new object with https://github.com/neo-project/neo/pull/1495/files#r396017725

Reply:
You can't reuse it. Otherwise when it is sent to other actors, and we may modify the same object at the same time.